### ISSUE TITLE
Kill DictValue, FailureValue, SkipValue. Rename NumericValue to Value

### ISSUE THREAD

**Issue by benshayden** [catapult-project/catapult#2650]
"We drastically over-engineered failure handling."

**Comment by eakuefner** [comment]
To delete `FailureValue`, I need to undertake the following steps:

1. Switch the frontend code to use `metric_map_function` rather than calling metrics directly (@nedn turned out to have been right all those months ago)
2. Use MRE `Failure` instances in those places where we currently lean on `FailureValue`
3. Delete `FailureValue` at last.

Broadly, it seems like we should make `ValueSet` (I think `MetricResult` in the TBMv2.2 world) a *subclass* of `MreResult`, so that the pairs are values by GUID, and the failures are first-class.

**Comment by benshayden** [comment]
:+1:

**Comment by eakuefner** [comment]
To use `Failure`, we've realized it would be most comfortable to just get it over with and move MRE to tracing. To do this, we basically need to delete PI; that effort is tracked in #2035 and I'm going to comment there about the steps I'm taking.

**Comment by benshayden** [comment]
We found a simpler way of making metric-side-panel display errors instead of routing them through value-set-table. Metric functions ought never produce FailureValues nor mre Failures.

We decided not to try to transfigure ValueSet into a subclass of MreResult. value-set-table wants to work with Histograms, but we can't put Histogram objects into MreResults because MreResult doesn't know how to serialize them, and shouldn't. metricMapFunction can't control whether it's given a MreResult or a MetricResult, and shouldn't have to act differently depending on what it gets. We tried a few different ways of making MetricResult work, and they just didn't sit right.

So let's keep ValueSet as is, unless somebody else can figure it out.

We'll still merge NumericValue into Histogram for ergonomics, delete NumericBase, and move ScalarNumeric to Scalar in base/ so that metrics don't have to think about them.

We'll move Histogram's statistics scalars from ValueSet to metricMapFunction, again so that metrics don't have to think about them. Let's never put a scalar in a ValueSet ever again, and burn buildFromSamples().

After telemetry uses a map_single_file mapper function to merge Histograms between story repetitions, then we'll move the statistics scalars from metricMapFunction to another map_single_file mapper function for telemetry to call after it merges Histograms between repetitions. Or maybe the dashboard will be ready to deal with Histograms by then and we can burn Histogram.getSummarizedScalarNumericsWithNames and Histogram.customizeSummaryOptions, or we can use a single mapper function to both merge and generate statistics scalars, or whatever.
